notifiers/slack/slack.go

```diff
@@ -0,0 +1,141 @@
+package slack
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+const (
+	Resolved  = "good"
+	Triggered = "warning"
+)
+
+var (
+	client = &http.Client{Timeout: 10 * time.Second}
+)
+
+type attachment struct {
+	Color string `json:"color,omitempty"`
+	Title string `json:"title"`
+	Text  string `json:"text"`
+}
+
+func (a attachment) String() string {
+	return fmt.Sprintf("{title:%s, text:%s, color:%s}", a.Title, a.Text, a.Color)
+}
+
+type request struct {
+	Channel     string        `json:"channel,omitempty"`
+	Username    string        `json:"username,omitempty"`
+	IconURL     string        `json:"icon_url,omitempty"`
+	IconEmoji   string        `json:"icon_emoji,omitempty"`
+	Attachments []*attachment `json:"attachments"`
+}
+
+type Client struct {
+	cfg      *Config
+	messages []*attachment
```

Comment:
Given we only support a single message at a time now, what's the reasoning for the multi-message abstraction here?